### ISSUE TITLE
More in depth guide please (many small questions)

### ISSUE THREAD

**Issue by ran88dom99** [EpistasisLab/tpot#728]
I can not find a guide on choosing TPOT parameters. I know the API is explained in the documents but its too brief. TPOT seems made for users unskilled in ML and GP.  Should I set population_size always less than offspring_size? What to do with small (130) datasets? Will larger or smaller cross-validation help? Can I add bagging? "We recommend using the default parameter unless you understand how the mutation rate affects GP algorithms. " should have a link.

Running TPOT for a day with the right parameters gets cv RMSE from 1 to almost 0 but .44 RMSE  on a holdout set. This is good but is there anything more I can do?

Can TPOT or any of its models detect conditional statements, especially nested ones? Such as if X1<0 then Y=X2 else Y=-X2. Only [cubist and mob seem to manage.](https://github.com/ran88dom99/GeneratedRegMLBenchmark/wiki)

**Comment by rhiever** [comment]
The requests wrt adding a more detailed user guide for TPOT are fair. @ran88dom99, can you please file a separate issue for improving the user guide? I can't guarantee that it will be a top priority, but let's at least get an issue filed for it.

To answer your specific questions:

>Should I set population_size always less than offspring_size?

In general, you want `population_size` to be at least 100, preferably 500+ if you have the time. It's likely not worth changing `offspring_size`. For my 'long' runs, I will often set `population_size` to 1000 and `generations` to 1000. You can also make use of the `early_stopping` parameter to end the TPOT run early if no improvements are being made.

>What to do with small (130) datasets?

Overfitting is a major concern here. I suggest using the 'TPOT light' configuration, as 130 records is likely not enough for the more complex methods used in the default TPOT configuration. Make sure to have a good test set to verify that TPOT is not overfitting on your training set.

>Running TPOT for a day with the right parameters gets cv RMSE from 1 to almost 0 but .44 RMSE on a holdout set. Anything more I can do?

The best solution is to collect more samples. Barring that, you can try the LeaveOneOut `cv` scheme and use the 'TPOT light' configuration if you suspect overfitting is an issue.
